hazelcast/src/main/java/com/hazelcast/query/impl/AndResultSet.java

```diff
@@ -28,37 +28,26 @@
 import java.util.Set;
 
 import static com.hazelcast.util.Preconditions.isNotNull;
+
 /**
  * And Result set for Predicates.
  */
 public class AndResultSet extends AbstractSet<QueryableEntry> {
     private final Set<QueryableEntry> setSmallest;
     private final List<Set<QueryableEntry>> otherIndexedResults;
     private final List<Predicate> lsNoIndexPredicates;
+    private int cachedSize;
 
     public AndResultSet(Set<QueryableEntry> setSmallest, List<Set<QueryableEntry>> otherIndexedResults,
                         List<Predicate> lsNoIndexPredicates) {
         this.setSmallest = isNotNull(setSmallest, "setSmallest");
         this.otherIndexedResults = otherIndexedResults;
         this.lsNoIndexPredicates = lsNoIndexPredicates;
+        this.cachedSize = -1;
     }
 
     public byte[] toByteArray(ObjectDataOutput out) throws IOException {
```

Comment:
That's a good hint. First I thought it overrides a method in higher-levelled interface. Now I see it's not used.
Should we delete it? Wdyt also @jerrinot @mdogan?